File: ArcadeManao/solve/ruby/ArcadeManaoSolver.rb
#! /usr/bin/env ruby

require_relative "ArcadeManao"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  level = reader.next(TArray.new(TString))
  reader.next()
  coinRow = reader.next(TInt)
  reader.next()
  coinColumn = reader.next(TInt)

  result = ArcadeManao.new().shortestLadder(level, coinRow, coinColumn)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main

File: ArcadeManao/solve/python/ArcadeManaoSolver.py
#! /usr/bin/python3

import os, sys
import ArcadeManao

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        level = reader.next("String[]")
        reader.next()
        coinRow = reader.next("int")
        reader.next()
        coinColumn = reader.next("int")

    result = ArcadeManao.shortestLadder(level, coinRow, coinColumn)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "int"))

if __name__ == "__main__":
    init()
    main()
